### ISSUE TITLE
[v3] Implement new Concept Exercise: Ranges

### ISSUE THREAD

**Issue by wneumann** [exercism/swift#413]
This issue describes how to implement the Ranges concept exercise for the Swift track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read the following documents:

- [The features of v3][docs-features-of-v3].
- [Rationale for v3][docs-rationale-for-v3].
- [What are concept exercises and how are they structured?][docs-concept-exercises]

Please also watch the following video:

- [The Anatomy of a Concept Exercise][anatomy-of-a-concept-exercise].

## Goal

This concept exercise should convey a basic understanding of how to `Range`s and `ClosedRange`s in Swift.

## Learning objective

- Know how to create a `Range`
- Know how to create a `ClosedRange`
- Know the difference between `Range` and `ClosedRange`
- Know how to pattern match `Range`s
- Know that `Ranges` can be used with types other than `Int` and what, e.g. a `ClosedRange<String>` means.
- Know how to iterate over a `Strideable` `Range` (perhaps just talk about `Ints and not bring up `Strideable`)

## Out of scope

- Mapping, filtering, reducing, etc.
- `Strideable` protocol

## Concepts

- `Range`
- `ClosedRange`
- …

## Prerequisites

- `conditionals`
- `numbers`
- `strings-and-characters`

### Hints

- [range operators][range-operators]
- [for-in loops][for-in-loops]
- [expression patterns][expression-pattern]

### After

- [range operators][range-operators]
- [for-in loops][for-in-loops]
- [expression patterns][expression-pattern]
- [closed range docs][closed-range-docs]
- [range docs][range-docs]

## Representer

TBD

## Analyzer

TBD

## Implementing

See [the Swift track implementation guide][swift-implementing-concept-exercises] for guidance.

##Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

##Edits

[swift-implementing-concept-exercises]: https://github.com/wneumann/v3/blob/master/languages/swift/reference/implementing-a-concept-exercise.md
[docs-concept-exercises]: ../../../docs/concept-exercises.md
[docs-rationale-for-v3]: ../../../docs/rationale-for-v3.md
[docs-features-of-v3]: ../../../docs/features-of-v3.md
[anatomy-of-a-concept-exercise]: https://www.youtube.com/watch?v=gkbBqd7hPrA

[range-operators]: https://docs.swift.org/swift-book/LanguageGuide/BasicOperators.html#ID73
[for-in-loops]: https://docs.swift.org/swift-book/LanguageGuide/ControlFlow.html#ID121
[expression-pattern]: https://docs.swift.org/swift-book/ReferenceManual/Patterns.html#ID426
[range-docs]: https://developer.apple.com/documentation/swift/range
[closed-range-docs]: https://developer.apple.com/documentation/swift/closedrange
